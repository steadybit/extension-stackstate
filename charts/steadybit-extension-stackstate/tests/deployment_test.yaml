templates:
  - deployment.yaml
tests:
  - it: manifest should match snapshot without TLS
    asserts:
      - matchSnapshot: { }
  - it: manifest should match snapshot with TLS
    set:
      tls:
        server:
          certificate:
            fromSecret: server-cert
    asserts:
      - matchSnapshot: { }
  - it: manifest should match snapshot with mutual TLS
    set:
      tls:
        server:
          certificate:
            fromSecret: server-cert
        client:
          certificates:
            fromSecrets:
              - client-cert-a
              - client-cert-a
    asserts:
      - matchSnapshot: {}
  - it: manifest should match snapshot with extra env vars
    set:
      extraEnv:
        - name: FOO
          value: "bar"
      extraEnvFrom:
        - configMapRef:
          name: env-configmap
        - secretRef:
          name: env-secrets
    asserts:
      - matchSnapshot: {}
  - it: manifest should match snapshot with env vars
    set:
      stackstate:
        apiBaseUrl: "https://stackstate.example.com"
        serviceToken: "123"
    asserts:
      - matchSnapshot: {}
  - it: manifest should match snapshot with existing secret
    set:
      stackstate:
        apiBaseUrl: "https://stackstate.example.com"
        existingSecret: "secret-stackstate-existing"
    asserts:
      - matchSnapshot: {}
