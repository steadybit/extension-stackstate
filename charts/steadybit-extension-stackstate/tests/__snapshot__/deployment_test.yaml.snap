manifest should match snapshot using podAnnotations and Labels:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        steadybit.com/discovery-enabled: "false"
      name: RELEASE-NAME-steadybit-extension-stackstate
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/name: steadybit-extension-stackstate
      template:
        metadata:
          annotations:
            some-annotation: some-annotation-value
          labels:
            app.kubernetes.io/name: steadybit-extension-stackstate
            some-label: some-label-value
            steadybit.com/discovery-enabled: "false"
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_SERVICE_TOKEN
                  valueFrom:
                    secretKeyRef:
                      key: service-token
                      name: steadybit-extension-stackstate
                - name: STEADYBIT_EXTENSION_API_BASE_URL
                  value: null
              image: ghcr.io/steadybit/extension-stackstate:latest
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /health/liveness
                  port: 8084
              name: extension
              readinessProbe:
                httpGet:
                  path: /health/readiness
                  port: 8084
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 64Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsGroup: 10000
                runAsNonRoot: true
                runAsUser: 10000
              volumeMounts: null
          volumes: null
manifest should match snapshot with TLS:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        steadybit.com/discovery-disabled: "true"
      name: RELEASE-NAME-steadybit-extension-stackstate
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/name: steadybit-extension-stackstate
      template:
        metadata:
          annotations: null
          labels:
            app.kubernetes.io/name: steadybit-extension-stackstate
            steadybit.com/discovery-enabled: "false"
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_TLS_SERVER_CERT
                  value: /etc/extension/certificates/server-cert/tls.crt
                - name: STEADYBIT_EXTENSION_TLS_SERVER_KEY
                  value: /etc/extension/certificates/server-cert/tls.key
                - name: STEADYBIT_EXTENSION_SERVICE_TOKEN
                  valueFrom:
                    secretKeyRef:
                      key: service-token
                      name: steadybit-extension-stackstate
                - name: STEADYBIT_EXTENSION_API_BASE_URL
                  value: null
              image: ghcr.io/steadybit/extension-stackstate:latest
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /health/liveness
                  port: 8084
              name: extension
              readinessProbe:
                httpGet:
                  path: /health/readiness
                  port: 8084
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 64Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsGroup: 10000
                runAsNonRoot: true
                runAsUser: 10000
              volumeMounts:
                - mountPath: /etc/extension/certificates/server-cert
                  name: certificate-server-cert
                  readOnly: true
          volumes:
            - name: certificate-server-cert
              secret:
                optional: false
                secretName: server-cert
manifest should match snapshot with env vars:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        steadybit.com/discovery-disabled: "true"
      name: RELEASE-NAME-steadybit-extension-stackstate
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/name: steadybit-extension-stackstate
      template:
        metadata:
          annotations: null
          labels:
            app.kubernetes.io/name: steadybit-extension-stackstate
            steadybit.com/discovery-enabled: "false"
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_SERVICE_TOKEN
                  valueFrom:
                    secretKeyRef:
                      key: service-token
                      name: steadybit-extension-stackstate
                - name: STEADYBIT_EXTENSION_API_BASE_URL
                  value: https://stackstate.example.com
              image: ghcr.io/steadybit/extension-stackstate:latest
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /health/liveness
                  port: 8084
              name: extension
              readinessProbe:
                httpGet:
                  path: /health/readiness
                  port: 8084
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 64Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsGroup: 10000
                runAsNonRoot: true
                runAsUser: 10000
              volumeMounts: null
          volumes: null
manifest should match snapshot with existing secret:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        steadybit.com/discovery-disabled: "true"
      name: RELEASE-NAME-steadybit-extension-stackstate
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/name: steadybit-extension-stackstate
      template:
        metadata:
          annotations: null
          labels:
            app.kubernetes.io/name: steadybit-extension-stackstate
            steadybit.com/discovery-enabled: "false"
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_SERVICE_TOKEN
                  valueFrom:
                    secretKeyRef:
                      key: service-token
                      name: secret-stackstate-existing
                - name: STEADYBIT_EXTENSION_API_BASE_URL
                  value: https://stackstate.example.com
              image: ghcr.io/steadybit/extension-stackstate:latest
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /health/liveness
                  port: 8084
              name: extension
              readinessProbe:
                httpGet:
                  path: /health/readiness
                  port: 8084
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 64Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsGroup: 10000
                runAsNonRoot: true
                runAsUser: 10000
              volumeMounts: null
          volumes: null
manifest should match snapshot with extra env vars:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        steadybit.com/discovery-disabled: "true"
      name: RELEASE-NAME-steadybit-extension-stackstate
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/name: steadybit-extension-stackstate
      template:
        metadata:
          annotations: null
          labels:
            app.kubernetes.io/name: steadybit-extension-stackstate
            steadybit.com/discovery-enabled: "false"
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_SERVICE_TOKEN
                  valueFrom:
                    secretKeyRef:
                      key: service-token
                      name: steadybit-extension-stackstate
                - name: STEADYBIT_EXTENSION_API_BASE_URL
                  value: null
                - name: FOO
                  value: bar
              envFrom:
                - configMapRef: null
                  name: env-configmap
                - name: env-secrets
                  secretRef: null
              image: ghcr.io/steadybit/extension-stackstate:latest
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /health/liveness
                  port: 8084
              name: extension
              readinessProbe:
                httpGet:
                  path: /health/readiness
                  port: 8084
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 64Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsGroup: 10000
                runAsNonRoot: true
                runAsUser: 10000
              volumeMounts: null
          volumes: null
manifest should match snapshot with extra labels:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        steadybit.com/discovery-disabled: "true"
        tags.datadoghq.com/service: steadybit-extension
      name: RELEASE-NAME-steadybit-extension-stackstate
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/name: steadybit-extension-stackstate
      template:
        metadata:
          labels:
            app.kubernetes.io/name: steadybit-extension-stackstate
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_SERVICE_TOKEN
                  valueFrom:
                    secretKeyRef:
                      key: service-token
                      name: steadybit-extension-stackstate
                - name: STEADYBIT_EXTENSION_API_BASE_URL
                  value: null
              image: ghcr.io/steadybit/extension-stackstate:latest
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /health/liveness
                  port: 8084
              name: extension
              readinessProbe:
                httpGet:
                  path: /health/readiness
                  port: 8084
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 64Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsGroup: 10000
                runAsNonRoot: true
                runAsUser: 10000
              volumeMounts: null
          volumes: null
manifest should match snapshot with mutual TLS:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        steadybit.com/discovery-disabled: "true"
      name: RELEASE-NAME-steadybit-extension-stackstate
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/name: steadybit-extension-stackstate
      template:
        metadata:
          annotations: null
          labels:
            app.kubernetes.io/name: steadybit-extension-stackstate
            steadybit.com/discovery-enabled: "false"
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_TLS_SERVER_CERT
                  value: /etc/extension/certificates/server-cert/tls.crt
                - name: STEADYBIT_EXTENSION_TLS_SERVER_KEY
                  value: /etc/extension/certificates/server-cert/tls.key
                - name: STEADYBIT_EXTENSION_TLS_CLIENT_CAS
                  value: /etc/extension/certificates/client-cert-a/tls.crt,/etc/extension/certificates/client-cert-a/tls.crt
                - name: STEADYBIT_EXTENSION_SERVICE_TOKEN
                  valueFrom:
                    secretKeyRef:
                      key: service-token
                      name: steadybit-extension-stackstate
                - name: STEADYBIT_EXTENSION_API_BASE_URL
                  value: null
              image: ghcr.io/steadybit/extension-stackstate:latest
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /health/liveness
                  port: 8084
              name: extension
              readinessProbe:
                httpGet:
                  path: /health/readiness
                  port: 8084
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 64Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsGroup: 10000
                runAsNonRoot: true
                runAsUser: 10000
              volumeMounts:
                - mountPath: /etc/extension/certificates/client-cert-a
                  name: certificate-client-cert-a
                  readOnly: true
                - mountPath: /etc/extension/certificates/server-cert
                  name: certificate-server-cert
                  readOnly: true
          volumes:
            - name: certificate-client-cert-a
              secret:
                optional: false
                secretName: client-cert-a
            - name: certificate-server-cert
              secret:
                optional: false
                secretName: server-cert
manifest should match snapshot with mutual TLS using containerPaths:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        steadybit.com/discovery-enabled: "false"
      name: RELEASE-NAME-steadybit-extension-stackstate
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/name: steadybit-extension-stackstate
      template:
        metadata:
          annotations: null
          labels:
            app.kubernetes.io/name: steadybit-extension-stackstate
            steadybit.com/discovery-enabled: "false"
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_TLS_SERVER_CERT
                  value: /etc/tls/server.crt
                - name: STEADYBIT_EXTENSION_TLS_SERVER_KEY
                  value: /etc/tls/server.key
                - name: STEADYBIT_EXTENSION_TLS_CLIENT_CAS
                  value: /etc/tls/ca.crt,/etc/tls/ca2.crt
                - name: STEADYBIT_EXTENSION_SERVICE_TOKEN
                  valueFrom:
                    secretKeyRef:
                      key: service-token
                      name: steadybit-extension-stackstate
                - name: STEADYBIT_EXTENSION_API_BASE_URL
                  value: null
              image: ghcr.io/steadybit/extension-stackstate:latest
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /health/liveness
                  port: 8084
              name: extension
              readinessProbe:
                httpGet:
                  path: /health/readiness
                  port: 8084
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 64Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsGroup: 10000
                runAsNonRoot: true
                runAsUser: 10000
              volumeMounts: null
          volumes: null
manifest should match snapshot without TLS:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        steadybit.com/discovery-disabled: "true"
      name: RELEASE-NAME-steadybit-extension-stackstate
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/name: steadybit-extension-stackstate
      template:
        metadata:
          annotations: null
          labels:
            app.kubernetes.io/name: steadybit-extension-stackstate
            steadybit.com/discovery-enabled: "false"
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_SERVICE_TOKEN
                  valueFrom:
                    secretKeyRef:
                      key: service-token
                      name: steadybit-extension-stackstate
                - name: STEADYBIT_EXTENSION_API_BASE_URL
                  value: null
              image: ghcr.io/steadybit/extension-stackstate:latest
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /health/liveness
                  port: 8084
              name: extension
              readinessProbe:
                httpGet:
                  path: /health/readiness
                  port: 8084
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 64Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsGroup: 10000
                runAsNonRoot: true
                runAsUser: 10000
              volumeMounts: null
          volumes: null
